### PR TITLE
kubectl: update message for a no-op patch

### PR DESCRIPTION
**Pull request by nikhita**
Fixes https://github.com/kubernetes/kubernetes/issues/67344

/cc liggitt juanvallejo soltysh timoreimann 
/sig cli

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by nikhita** [comment]
cc @timoreimann (since you had worked on the dry run option for patch https://github.com/kubernetes/kubernetes/pull/60675)

File: pkg/kubectl/cmd/patch.go [hunk 1]
```diff
@@ -331,5 +331,5 @@ func patchOperation(didPatch bool) string {
 	if didPatch {
 		return "patched"
 	}
-	return "not patched"
+	return "not patched (no change)"
```
Comment (liggitt) [inline pkg/kubectl/cmd/patch.go]: I still think "not patched" is still misleading when a patch was actually submitted to the server... "patched (no change)" seems more accurate to me

Comment (nikhita) [reply pkg/kubectl/cmd/patch.go]: > I still think "not patched" is misleading when a patch was actually submitted to the server... "patched (no change)" seems more accurate to me

agreed. done.

**Review by juanvallejo — APPROVED** [review]

**Comment by juanvallejo** [comment]
lgtm

**Pull request merged**